src/extension.ts

```diff
@@ -18,191 +18,238 @@ import { Utilities } from "./utilities";
 import { ResultsListController } from "./resultsListController";
 import { FileMapper } from "./fileMapper";
 import { SVDiagnosticCollection } from "./svDiagnosticCollection";
-import { Api } from "./api/sarifViewerApi";
+import { OpenLogArguments, Api } from "./api/sarifViewerApi";
 
 // This is equivalent to "including" the generated javascript to get the code to run that sets the prototypes for the extension methods.
 // If you don't do this... you crash using the extension methods.
 import './utilities/stringUtilities';
-import { ApiImpl } from './utilities/apiImpl';
 
-/**
- * This method is called when the extension is activated.
- * Creates the explorer, reader, provider
- * Process any open SARIF Files
- */
 export function activate(context: vscode.ExtensionContext): Api {
-    Utilities.initialize(context);
-    FileConverter.initialize(context);
-
-    context.subscriptions.push(FileMapper.InitializeFileMapper());
-
-    const diagnosticCollection: SVDiagnosticCollection = new SVDiagnosticCollection();
-    context.subscriptions.push(diagnosticCollection);
-
-    const explorerController: ExplorerController = new ExplorerController(context, diagnosticCollection);
-    context.subscriptions.push(explorerController);
-
-    const codeActionProvider: SVCodeActionProvider = new SVCodeActionProvider(diagnosticCollection);
-    context.subscriptions.push(codeActionProvider);
-
-    context.subscriptions.push(new ResultsListController(explorerController, codeActionProvider, diagnosticCollection));
-
-    context.subscriptions.push(new CodeFlowCodeLensProvider(explorerController, diagnosticCollection));
-
-    context.subscriptions.push(new CodeFlowDecorations(explorerController, diagnosticCollection));
-
-    const logReader: LogReader = new LogReader();
-    context.subscriptions.push(logReader);
-
-    // Listen for new sarif files to open or close
-    context.subscriptions.push(vscode.workspace.onDidOpenTextDocument(async (textDocument) => {
-        await onDocumentOpened(textDocument, logReader, diagnosticCollection);
-    }));
-
-    context.subscriptions.push(vscode.workspace.onDidCloseTextDocument((textDocument) => {
-        onDocumentClosed(textDocument, diagnosticCollection);
-    }));
-
-    // We do not need to block extension startup for reading any open documents.
-    void readOpenedDocuments(logReader, diagnosticCollection);
-
-    return new ApiImpl(logReader, diagnosticCollection);
+    return new SarifExtension(context);
 }
 
 /**
  * Clean up extension if it gets deactivated
  */
 export function deactivate(): void {
```

Comment:
>deactivate [](start = 16, length = 10)

Verify that this disposes the extension properly.

Reply:
Confirmed. Everything disposes correctly.

---
In reply to: [412425847](https://github.com/microsoft/sarif-vscode-extension/pull/253#discussion_r412425847) [](ancestors = 412425847)